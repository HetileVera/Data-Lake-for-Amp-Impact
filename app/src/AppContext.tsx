import React, { createContext, useContext } from "react";
import isEmpty from "lodash/isEmpty";
import throttle from "lodash/throttle";

import { AppStore, IAppStore } from "./models/AppStore";
import { IInstallation } from "./models/Installation";
import { isLocalDev, websiteUrlOrigin, assetWebsiteUrl, secretKey } from "./helpers/settings";
import { onSnapshot } from "mobx-state-tree";

// Create an instance of the root store (a.k.a appStore)
const appStore = AppStore.create({});

// Store state to local storage
const storeLocally = throttle(() => appStore.saveLocal(), 5000);

// Store state in S3 when applicable
const storeRemotely = throttle(() => appStore.saveRemote(), 30000);

// The initialization function for the root store
export function initializeStore() {
  if (!isEmpty(secretKey)) console.warn("Access key is bundled!");

  if (isLocalDev) {
    console.log(`isLocalDev '${isLocalDev}'`);
    console.log(`websiteUrlOrigin '${websiteUrlOrigin}'`);
    console.log(`assetWebsiteUrl '${assetWebsiteUrl}'`);

    if (window.location.host !== "localhost:3000") {
      console.warn("You should run the app on localhost!");

      throw new Error("App should run on localhost for development mode!");
    }
  }

  appStore.loadLocal();
  onSnapshot(appStore, (snapshot) => {
    storeLocally();
    storeRemotely();
  });
}

// Create the store context
export const StoreContext = createContext<IAppStore | undefined>(undefined);

// Create the store provider component
export const StoreProvider: React.FunctionComponent = ({ children }) => {
  return <StoreContext.Provider value={appStore}>{children}</StoreContext.Provider>;
};

// useStore hook
export function useStore(): IAppStore {
  const store = useContext(StoreContext);

  if (!store) {
    throw new Error("useStore() must be used within a StoreProvider");
  }

  return store;
}

// useInstallation hook
export function useInstallation(): IInstallation {
  const store = useStore();
  const installation = store.installation;

  if (!installation) {
    throw new Error("An installation instance was not created yet");
  }

  return installation;
}
