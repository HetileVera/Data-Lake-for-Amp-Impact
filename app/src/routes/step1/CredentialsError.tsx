import { FC } from "react";
import { observer } from "mobx-react";
import { Alert, AlertDescription, AlertIcon, AlertTitle, Box, ListItem, UnorderedList } from "@chakra-ui/react";

import { CredentialsValidationErrorCode, CredentialsValidationException } from "../../api/validate-credentials";

export const CredentialsError: FC<{ exception?: CredentialsValidationException | Error }> = observer(({ exception }) => {
  if (!exception) return null;

  let title: string;
  let content: React.ReactElement = <></>;

  if (exception instanceof CredentialsValidationException) {
    title = "A problem with the access keys";
    switch (exception?.code) {
      case CredentialsValidationErrorCode.InvalidKeys:
        content = (
          <Box>
            The keys are invalid. This could be due to one or more of the following reasons:
            <UnorderedList ml={5} mt={2}>
              <ListItem>The keys are incorrect, check that you are using the correct values</ListItem>
              <ListItem>The access keys are deactivated</ListItem>
              <ListItem>The access keys are deleted and therefore no longer valid</ListItem>
            </UnorderedList>
          </Box>
        );
        break;
      case CredentialsValidationErrorCode.NotAdmin:
        content = (
          <Box>
            The keys are valid but they are for a user who does not have admin permissions. Add the <b>AdministratorAccess</b> policy to the user and
            try again.
          </Box>
        );
        break;
      case CredentialsValidationErrorCode.AccountMismatch:
        content = <Box>The keys are for a different AWS account.</Box>;
        break;

      default:
        content = (
          <Box>
            We are unable to validate the keys. This could be due to one or more of the following reasons:
            <UnorderedList ml={5} mt={2} mb={3}>
              <ListItem>A problem with the internet connection</ListItem>
              <ListItem>An intermittent problem while trying to validate the keys</ListItem>
            </UnorderedList>
            Try again in a few minutes
          </Box>
        );
        break;
    }
  } else {
    title = "A problem has been encountered";
    content = (
      <Box>
        We are unable to proceed to the next step. This could be due to one or more of the following reasons:
        <UnorderedList ml={5} mt={2} mb={3}>
          <ListItem>A problem with the internet connection</ListItem>
          <ListItem>An intermittent problem while trying to proceed to the next step</ListItem>
        </UnorderedList>
        Try again in a few minutes
      </Box>
    );
  }

  return (
    <Alert status="error" variant="left-accent" mt={5} color="red.700" alignItems="flex-start" borderColor="red.600">
      <AlertIcon mt="3px" color="red.600" />
      <Box>
        <AlertTitle mr={2}>{title}</AlertTitle>
        <AlertDescription>{content}</AlertDescription>
      </Box>
    </Alert>
  );
});
