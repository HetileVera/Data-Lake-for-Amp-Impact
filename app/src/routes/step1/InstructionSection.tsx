import React from "react";
import { Box, Divider, Stack, Link as ChakraLink, UnorderedList, ListItem } from "@chakra-ui/react";
import { ExternalLinkIcon } from "@chakra-ui/icons";

import awsConnectImage01 from "images/aws-connect-01.png";
import awsConnectImage02 from "images/aws-connect-02.png";

import { ClickableImage } from "components/ClickableImage";

export const InstructionSection = () => {
  return (
    <Box bg="orange.75" p={6} borderRadius="lg" border="1px" borderColor="orange.100">
      <Stack direction={{ base: "column", md: "row" }} spacing="30px" pt={6} pb={6}>
        <Box>
          <ClickableImage w={{ base: "full", md: "300px" }} src={awsConnectImage01} title="AWS Console Login Page" />
        </Box>
        <Box>
          <UnorderedList ml={5} mt={0} mb={3} color="orange.900">
            <ListItem>
              Log in to the &nbsp;
              <ChakraLink href="https://aws.amazon.com/console/" isExternal color="orange.600">
                AWS console <ExternalLinkIcon mx="2px" />
              </ChakraLink>
              &nbsp; using an admin user, not the root user.
            </ListItem>
          </UnorderedList>
          <Box fontSize="sm" mt={4} bg="orange.100" p={3} borderRadius="lg">
            Remember to log in using an admin user and not the root user
          </Box>
        </Box>
      </Stack>

      <Divider borderColor="orange.200" />

      <Stack direction={{ base: "column", md: "row" }} spacing="30px" mt={6} mb={0}>
        <Box>
          <ClickableImage w={{ base: "full", md: "300px" }} src={awsConnectImage02} title="AWS Console IAM Page" />
        </Box>
        <Box>
          <UnorderedList ml={5} mt={0} mb={0} color="orange.900">
            <ListItem>
              Then, head over to the{" "}
              <ChakraLink href="https://console.aws.amazon.com/iamv2/home#/users" isExternal color="orange.600">
                Identity and Management section (IAM) <ExternalLinkIcon mx="2px" />
              </ChakraLink>
            </ListItem>
            <ListItem mt={4}>
              Pick the user name that you used to log in. The user should have admin permissions. Then, select the Security Credentials tab.
            </ListItem>
            <ListItem mt={4}>
              Click on the Create access key button to generate the access keys, you will get a chance to view and download the access keys. You can
              now use the access keys to fill in the information below.
            </ListItem>
          </UnorderedList>
        </Box>
      </Stack>
    </Box>
  );
};
