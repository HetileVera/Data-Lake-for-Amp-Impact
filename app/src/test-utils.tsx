import React from "react";
import { render, RenderOptions } from "@testing-library/react";
import { ChakraProvider } from "@chakra-ui/react";

import { theme } from "themes/orange";

export const CustomChakraProvider = ({ children }: { children: React.ReactNode }) => {
  return (
    <ChakraProvider theme={theme}>
      <div data-testid="test-root">{children}</div>
    </ChakraProvider>
  );
};

const customRender = (ui: React.ReactElement, options?: Omit<RenderOptions, "wrapper">) => render(ui, { wrapper: CustomChakraProvider, ...options });

export * from "@testing-library/react";

export { customRender as render };
