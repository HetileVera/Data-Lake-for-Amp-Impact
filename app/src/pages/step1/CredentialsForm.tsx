import { FC, useState } from "react";
import isEmpty from "lodash/isEmpty";
import { Box, Heading, Button, FormErrorMessage, FormControl, Input, Alert, AlertIcon, InputGroup, InputRightElement } from "@chakra-ui/react";
import { UseFormRegister, FieldErrors, useWatch, Control } from "react-hook-form";

import { useStore } from "AppContext";
import { useColorScheme } from "models/useColorScheme";
import { accessKey, secretKey } from "helpers/settings";

export interface ICredentialsFormInput {
  accessKeyId: string;
  secretAccessKey: string;
  region: string;
}
export interface ICredentialsForm {
  register: UseFormRegister<ICredentialsFormInput>;
  errors: FieldErrors<ICredentialsFormInput>;
  isSubmitting: boolean;
  control: Control<ICredentialsFormInput>;
}

export const CredentialsForm = ({ register, errors, isSubmitting, control }: ICredentialsForm) => {
  const store = useStore();
  const { tone, colorScheme } = useColorScheme();
  const credentials = store.credentials;
  const accessKeyDefaultValue = credentials.accessKey || accessKey;
  const secretKeyDefaultValue = credentials.secretKey || secretKey;
  const secretKeyValue = useWatch({ control, name: "secretAccessKey", defaultValue: secretKeyDefaultValue });
  const [show, setShow] = useState(false);
  const handleClick = () => setShow(!show);

  return (
    <Box p={6} borderRadius="lg" boxShadow="base" bg={tone(50)}>
      <Heading size="sm" pt="5px" pb="20px" color={tone(600)}>
        Access Key ID &amp; Secret Access Key
      </Heading>

      {/* @ts-ignore */}
      <FormControl isInvalid={errors.accessKeyId || errors.secretAccessKey}>
        <Input
          id="accessKeyId"
          placeholder="Access key id"
          defaultValue={accessKeyDefaultValue}
          {...register("accessKeyId", { required: "Required" })}
          focusBorderColor={tone(700)}
          bg={tone(75)}
          disabled={isSubmitting}
        />
        <FormErrorMessage>{errors.accessKeyId && errors.accessKeyId.message}</FormErrorMessage>

        <InputGroup size="md" mt={5}>
          <Input
            pr="4.5rem"
            type={show ? "text" : "password"}
            placeholder="Secret access key"
            id="secretAccessKey"
            defaultValue={secretKeyDefaultValue}
            {...register("secretAccessKey", { required: "Required" })}
            focusBorderColor={tone(700)}
            bg={tone(75)}
            disabled={isSubmitting}
          />
          <InputRightElement width="4.5rem">
            <Button
              h="1.75rem"
              size="sm"
              onClick={handleClick}
              colorScheme={colorScheme}
              variant="outline"
              _focus={{ boxShadow: "none" }}
              isDisabled={isEmpty(secretKeyValue)}
            >
              {show ? "Hide" : "Show"}
            </Button>
          </InputRightElement>
        </InputGroup>

        <FormErrorMessage>{errors.secretAccessKey && errors.secretAccessKey.message}</FormErrorMessage>
        <Alert status="warning" variant="left-accent" mt={5} fontSize="sm">
          <AlertIcon />
          Secret access keys are sensitive information, keep them secured and protected, never post them on public platforms or leave them unsecured
          as this can compromise your account security
        </Alert>
      </FormControl>
    </Box>
  );
};
