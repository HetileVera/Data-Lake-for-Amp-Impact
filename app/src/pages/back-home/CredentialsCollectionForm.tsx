import React from "react";
import { Box, Button } from "@chakra-ui/react";
import { ArrowForwardIcon } from "@chakra-ui/icons";
import { observer } from "mobx-react";
import { SubmitHandler, useForm } from "react-hook-form";

import { useInstallation } from "AppContext";
import { useColorScheme } from "models/useColorScheme";
import { CredentialsValidationException } from "api/validate-credentials";

import { CredentialsForm, ICredentialsFormInput } from "pages/step1/CredentialsForm";
import { CredentialsError } from "pages/step1/CredentialsError";

interface ICredentialsCollectionForm {
  onCancel: () => void;
  onDone: () => void;
  buttonTitle?: string;
}

export const CredentialsCollectionForm = observer(({ onCancel, onDone, buttonTitle = "Resume" }: ICredentialsCollectionForm) => {
  const { colorScheme } = useColorScheme();
  const [error, setError] = React.useState<CredentialsValidationException | Error>();
  const installation = useInstallation();

  const onSubmit: SubmitHandler<ICredentialsFormInput> = async (values) => {
    setError(undefined);
    const accessKey = values.accessKeyId;
    const secretKey = values.secretAccessKey;
    const region = installation.region;

    try {
      await installation.connectToAwsStep.connectToAws(accessKey, secretKey, region);
      if (installation.isPostDeployment) installation.metadataStore.reset();

      return onDone();
    } catch (err: any) {
      console.log(err);
      setError(err);
    }
  };

  const {
    handleSubmit,
    register,
    control,
    formState: { errors, isSubmitting },
  } = useForm<ICredentialsFormInput>();

  const buttonProps: any = {};
  if (buttonTitle === "Resume") buttonProps.rightIcon = <ArrowForwardIcon />;

  return (
    <>
      <CredentialsError exception={error} />

      <Box pt={6}>
        {/* @ts-ignore */}
        <CredentialsForm {...{ register, errors, isSubmitting, control }} />
      </Box>

      <Box textAlign="right" w="full" mt={6} mb={0}>
        <Button colorScheme={colorScheme} size="md" disabled={isSubmitting} variant="ghost" onClick={() => onCancel()}>
          Cancel
        </Button>
        <Button
          colorScheme={colorScheme}
          size="md"
          loadingText="Connecting"
          ml={3}
          isLoading={isSubmitting}
          disabled={isSubmitting}
          onClick={handleSubmit(onSubmit)}
          {...buttonProps}
        >
          {buttonTitle}
        </Button>
      </Box>
    </>
  );
});
