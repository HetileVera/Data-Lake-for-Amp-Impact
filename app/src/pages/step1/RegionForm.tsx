import React from "react";
import { FieldErrors, UseFormRegister } from "react-hook-form";
import { Box, FormControl, FormErrorMessage, Heading, Select, Text } from "@chakra-ui/react";

import { ICredentialsFormInput } from "./CredentialsForm";
import { awsRegions } from "../../data/aws-regions";
import { useColorScheme } from "../../models/useColorScheme";

interface IRegionForm {
  register: UseFormRegister<ICredentialsFormInput>;
  errors: FieldErrors;
  isSubmitting: boolean;
  description: string;
  defaultRegion: string;
}

export const RegionForm = ({ register, errors, isSubmitting, description, defaultRegion }: IRegionForm) => {
  const { tone } = useColorScheme();

  return (
    <Box p={6} bg={tone(50)} borderRadius="lg" boxShadow="base">
      <Heading size="sm" pt="5px" pb="5px" color={tone(600)}>
        AWS Region
      </Heading>
      <Text fontSize="sm" color={tone(700)} mb={3}>
        {description}
      </Text>
      {/* @ts-ignore */}
      <FormControl isInvalid={errors.region || errors.region}>
        <Select bg={tone(75)} id="region" {...register("region", { required: "Required" })} defaultValue={defaultRegion} disabled={isSubmitting}>
          {awsRegions.map((item) => (
            <option key={item.name} className={item.name} value={item.name}>
              {item.label}
            </option>
          ))}
        </Select>
        <FormErrorMessage>{errors.region && errors.region.message}</FormErrorMessage>
      </FormControl>
    </Box>
  );
};
