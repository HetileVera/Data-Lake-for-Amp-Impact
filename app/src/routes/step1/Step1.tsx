import React from "react";
import { Box, Container, Heading, HStack, Button, Collapse, Stack } from "@chakra-ui/react";
import { ArrowForwardIcon, ArrowBackIcon } from "@chakra-ui/icons";
import { observer } from "mobx-react";
import { useForm, SubmitHandler } from "react-hook-form";
import { useNavigate } from "react-router-dom";

import { theme } from "themes/orange";
import { useInstallation, useStore } from "AppContext";
import { CredentialsValidationException } from "api/validate-credentials";
import { CurvedBox } from "components/CurvedBox";
import { StepsIndicator } from "components/StepsIndicator";
import { OutlineButton } from "components/OutlineButton";
import { CredentialsForm, ICredentialsFormInput } from "routes/step1/CredentialsForm";
import { RegionForm } from "routes/step1/RegionForm";
import { CredentialsError } from "routes/step1/CredentialsError";
import { YesNoAnswer } from "models/steps/BaseStep";
import { InstructionSection } from "routes/step1/InstructionSection";
import { Header } from "components/Header";

export const Step1 = observer(() => {
  const installation = useInstallation();
  const step = installation.connectToAwsStep;
  const needsAssistance = step.needsAssistance;
  const navigate = useNavigate();
  const [stepException, setStepException] = React.useState<CredentialsValidationException | Error>();
  const appStore = useStore();

  const handleAnswer = (answer: YesNoAnswer) => () => {
    step.setNeedsAssistance(answer);
  };

  const handlePrevious = () => {
    navigate("/");
  };

  const onSubmit: SubmitHandler<ICredentialsFormInput> = async (values) => {
    setStepException(undefined);
    const accessKey = values.accessKeyId;
    const secretKey = values.secretAccessKey;
    const region = values.region;
    try {
      await step.connectToAws(accessKey, secretKey, region);
      step.markCompleted();

      navigate("/steps/2");
    } catch (err: any) {
      console.log(err);
      setStepException(err);
    }
  };

  React.useEffect(() => {
    step.markStarted();
    window.scrollTo(0, 0);
  }, [appStore, step]);

  const {
    handleSubmit,
    register,
    control,
    formState: { errors, isSubmitting },
  } = useForm<ICredentialsFormInput>();

  const regionDescription =
    "AWS has many data centers grouped by geographical regions. Select a region that is closest to your location. We will keep your data in the region you select. If you are not sure, you can select the US East (Ohio) Region";

  return (
    <>
      <CurvedBox bgGradient={theme.gradients.bgLight} />

      <Header />

      <Box position="relative">
        <CurvedBox />
        <Box bg="orange.400" w="full" position="relative">
          <Container maxW="container.md" pt="15px" pb="15px">
            <StepsIndicator current={1} />
          </Container>
        </Box>

        <Container maxW="container.md" pt="0px" position="relative">
          <Heading display="inline-block" size="lg" pt="16px" pb="30px" color="gray.100" letterSpacing="-1px">
            Connect to your AWS account
          </Heading>

          <Box color="gray.50" mb={10}>
            <Stack direction="column" spacing="8px" align="left" justifyContent="space-between">
              <Box>
                We need your AWS <b>admin</b> credentials, specifically the <b>access keys</b>, so that we can connect to your AWS account and
                provision the resources for the data lake.
              </Box>
              <Box>
                Access keys consist of two parts: an access key id and a secret access key. These access keys must be for a user with admin
                permissions.
              </Box>
              <Box>If you haven't created access keys before, we will guide you through the process of obtaining the needed credentials.</Box>
            </Stack>
          </Box>

          <Box borderRadius="lg" boxShadow="base" bg="orange.50" mt={0} p={6}>
            <Box p={0}>
              <Heading size="md" pt="0px" pb="30px" color="orange.600" letterSpacing="-1px">
                Would you need assistance in creating AWS admin access and secret keys?
              </Heading>
              <Box textAlign="right" w="full" mb={4}>
                <OutlineButton selected={needsAssistance === YesNoAnswer.Yes} onClick={handleAnswer(YesNoAnswer.Yes)}>
                  Yes
                </OutlineButton>
                <OutlineButton selected={needsAssistance === YesNoAnswer.No} onClick={handleAnswer(YesNoAnswer.No)} ml={3}>
                  No
                </OutlineButton>
              </Box>
            </Box>

            <Collapse in={needsAssistance === YesNoAnswer.Yes} animateOpacity>
              <Box pt={6}>
                <InstructionSection />
              </Box>
            </Collapse>
          </Box>

          <Collapse in={needsAssistance !== YesNoAnswer.MissingAnswer} animateOpacity>
            <CredentialsError exception={stepException} />

            <Box pt={6}>
              {/* @ts-ignore */}
              <CredentialsForm {...{ register, errors, isSubmitting, control }} />
            </Box>

            <Box pt={6}>
              {/* @ts-ignore */}
              <RegionForm {...{ register, errors, isSubmitting }} description={regionDescription} defaultRegion={installation.region} />
            </Box>
          </Collapse>

          <HStack justifyContent="space-between" p={3} pt={6} mb={12}>
            <Button colorScheme="orange" size="md" leftIcon={<ArrowBackIcon />} disabled={isSubmitting} variant="ghost" onClick={handlePrevious}>
              Previous
            </Button>

            <Button
              colorScheme="orange"
              size="md"
              loadingText="Connecting"
              rightIcon={<ArrowForwardIcon />}
              ml={3}
              isLoading={isSubmitting}
              disabled={needsAssistance === YesNoAnswer.MissingAnswer || isSubmitting}
              onClick={handleSubmit(onSubmit)}
            >
              Next
            </Button>
          </HStack>
        </Container>
      </Box>
    </>
  );
});
