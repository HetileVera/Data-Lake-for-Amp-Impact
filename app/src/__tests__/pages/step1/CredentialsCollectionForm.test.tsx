import { cleanup, screen } from "@testing-library/react";
import { create } from "react-test-renderer";

import { render, RenderWithChakra } from "test-utils";

import { StoreProvider } from "AppContext";

import { CredentialsCollectionForm } from "pages/step1/CredentialsCollectionForm";

const mockOnCancel = jest.fn();
const mockOnDone = jest.fn();

describe("Step1 -> CredentialsCollectionForm component", () => {
  afterEach(cleanup);

  test("snapshot", () => {
    const tree = create(
      <RenderWithChakra>
        <StoreProvider>
          <CredentialsCollectionForm onCancel={mockOnCancel} onDone={mockOnDone} />
        </StoreProvider>
      </RenderWithChakra>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test("render properly", () => {
    render(
      <StoreProvider>
        <CredentialsCollectionForm onCancel={mockOnCancel} onDone={mockOnDone} />
      </StoreProvider>
    );

    expect(screen.getByText("Cancel")).toBeInTheDocument();
    expect(screen.getByText("Continue")).toBeInTheDocument();
  });
});
