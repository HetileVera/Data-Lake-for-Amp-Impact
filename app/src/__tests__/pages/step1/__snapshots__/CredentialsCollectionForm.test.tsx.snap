// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Step1 -> CredentialsCollectionForm component snapshot 1`] = `
Array [
  <div
    className="css-4lei0w"
  >
    <div
      className="css-19alo4x"
    >
      <h2
        className="chakra-heading css-1s3g4fq"
      >
        Access Key ID & Secret Access Key
      </h2>
      <div
        className="chakra-form-control css-1kxonj9"
        role="group"
      >
        <input
          className="chakra-input css-1rvlvqs"
          defaultValue=""
          disabled={false}
          id="accessKeyId"
          name="accessKeyId"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Access key id"
          readOnly={false}
          required={false}
        />
        <div
          className="chakra-input__group css-yymdw1"
        >
          <input
            className="chakra-input css-1h1dmm0"
            defaultValue=""
            disabled={false}
            id="secretAccessKey"
            name="secretAccessKey"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Secret access key"
            readOnly={false}
            required={false}
            type="password"
          />
          <div
            className="chakra-input__right-element css-ohd79a"
            size="md"
          >
            <button
              className="chakra-button css-1qo2nbo"
              disabled={true}
              onClick={[Function]}
              type="button"
            >
              Show
            </button>
          </div>
        </div>
        <div
          className="chakra-alert css-7lx40g"
          role="alert"
        >
          <span
            className="chakra-alert__icon css-15539z7"
          >
            <svg
              className="chakra-icon css-fi0ww0"
              focusable={false}
              viewBox="0 0 24 24"
            >
              <path
                d="M11.983,0a12.206,12.206,0,0,0-8.51,3.653A11.8,11.8,0,0,0,0,12.207,11.779,11.779,0,0,0,11.8,24h.214A12.111,12.111,0,0,0,24,11.791h0A11.766,11.766,0,0,0,11.983,0ZM10.5,16.542a1.476,1.476,0,0,1,1.449-1.53h.027a1.527,1.527,0,0,1,1.523,1.47,1.475,1.475,0,0,1-1.449,1.53h-.027A1.529,1.529,0,0,1,10.5,16.542ZM11,12.5v-6a1,1,0,0,1,2,0v6a1,1,0,1,1-2,0Z"
                fill="currentColor"
              />
            </svg>
          </span>
          Secret access keys are sensitive information, keep them secured and protected, never post them on public platforms or leave them unsecured as this can compromise your account security
        </div>
      </div>
    </div>
  </div>,
  <div
    className="css-s6koyw"
  >
    <button
      className="chakra-button css-7c7vkr"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Cancel
    </button>
    <button
      className="chakra-button css-zb27lh"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Continue
    </button>
  </div>,
  <span />,
]
`;
